src/persist/src/indexed/future.rs

```diff
@@ -162,13 +223,63 @@ impl<K: Data, V: Data> BlobFuture<K, V> {
         Ok(())
     }
 
-    /// Remove all batches containing only data strictly below the Future's time
+    /// Remove all batches containing only data strictly before the Future's time
```

Comment:
this is making me realize that before also kinda sounds like "in advance of". wonder if we should remove it from src/persist and replace it with something more descriptive. just musing, nothing for you to do here